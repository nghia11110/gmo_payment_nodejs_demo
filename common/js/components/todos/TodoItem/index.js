import React from 'react';
import PropTypes from 'prop-types';
import { List, Checkbox, Button } from 'semantic-ui-react';
import classnames from 'classnames/bind';
import css from './index.scss';

const cx = classnames.bind(css);

const TodoItem = props => {
  const { onRemove, onChange, todo: { id, completed, text } } = props;

  return (
    <List.Item className={classnames(css.todo, css.extra)}>
      <List.Content floated="right">
        <Button onClick={() => onRemove(id)} icon="remove" size="mini" />
      </List.Content>
      <List.Content floated="left">
        <Checkbox
          type="checkbox"
          checked={completed}
          onChange={() => onChange(id)}
        />
      </List.Content>
      <List.Content className={cx(css.text, { [css.completed]: completed })}>
        {text}
      </List.Content>
    </List.Item>
  );
};

TodoItem.propTypes = {
  todo: PropTypes.object.isRequired,
  onRemove: PropTypes.func,
  onChange: PropTypes.func
};

TodoItem.defaultProps = {
  onRemove: () => {},
  onChange: () => {}
};

export default TodoItem;
