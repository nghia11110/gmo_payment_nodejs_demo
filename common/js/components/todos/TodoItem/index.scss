.todo {
  &.extra {
    padding: 10px;
  }

  .completeInput {}

  .text {
    &.completed {
      text-decoration: line-through;
      color: #ccc;
    }
  }

  .delete {
    cursor: pointer;
    color: blue;
    text-decoration: underline
  }
}
