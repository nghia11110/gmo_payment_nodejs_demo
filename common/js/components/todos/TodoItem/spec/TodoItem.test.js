import React from 'react';
import TodoItem from '../index';
import { shallow } from 'enzyme';

describe('TodoItem', () => {
  it('renders correctly', () => {
    const component = shallow(<TodoItem todo={{ id: 1, text: 'Work' }} />);
    expect(component).toMatchSnapshot();
  });
});
