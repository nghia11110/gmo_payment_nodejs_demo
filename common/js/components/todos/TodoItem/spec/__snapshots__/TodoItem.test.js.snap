// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders correctly 1`] = `
<ListItem
  className="todo extra"
>
  <ListContent
    floated="right"
  >
    <Button
      as="button"
      icon="remove"
      onClick={[Function]}
      size="mini"
    />
  </ListContent>
  <ListContent
    floated="left"
  >
    <Checkbox
      onChange={[Function]}
      type="checkbox"
    />
  </ListContent>
  <ListContent
    className="text"
  >
    Work
  </ListContent>
</ListItem>
`;
